modules/setup/library/Setup/WebWizard.php

```diff
@@ -711,7 +709,23 @@ public function getRequirements($skipModules = false)
                 'setup.requirement.class'
             )
         )));
-        $set->merge($pgsqlSet);
+        $dbSet->merge($pgsqlSet);
+        $set->merge($dbSet);
+
+        $dbRequire = new RequirementSet(true);
+        $dbRequire->add(new SetRequirement(array(
+            'optional'      => $dbSet->getState(),
```

Comment:
Why is the optional state dependent on the set's state? This requirement should always be mandatory (`'optional' => false`)